src/test/java/org/example/FilteringAndSearchingTest.java

```diff
@@ -0,0 +1,208 @@
+
+        boolean anyTitleContainsInputWord = verifyEveryTitleContainsBrandName(brandName);
+
+        Assert.assertTrue(anyTitleContainsInputWord, "Not every title contains chosen brand name");
+    }
+
+    @DataProvider(name = "brandNamesAndPriceRanges")
+    public Object[][] brandNamesAndPriceRanges() {
+        return new Object[][]{
+                {"Corsair", 50, 250}
+        };
+    }
+
+    @Test(dataProvider = "brandNamesAndPriceRanges")
+    public void verifyProductsPricesAreInDefinedRange(String brandName, float minPrice, float maxPrice) {
+        selectBrand(brandName);
+
+        WebElement minPriceInput = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                .until(ExpectedConditions.presenceOfElementLocated(By.id("low-price")));
+        minPriceInput.sendKeys(String.valueOf(minPrice));
+
+        WebElement maxPriceInput = webDriver.findElement(By.id("high-price"));
+        maxPriceInput.sendKeys(String.valueOf(maxPrice));
+
+        WebElement submitPriceRangeBtn = webDriver.findElement(By.xpath("//input[@aria-labelledby='a-autoid-1-announce']"));
+        submitPriceRangeBtn.click();
+
+        boolean arePricesInChosenRange = verifyPricesAreInChoseRange(minPrice, maxPrice);
+
+        Assert.assertTrue(arePricesInChosenRange, "Prices are not within specified range");
+    }
+
+    @Test(dataProvider = "brandNames")
+    public void verifyProductsPricesAreSortedAscendingly(String brandName) {
+        selectBrand(brandName);
+
+        WebElement sortingDropdownList = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.id("a-autoid-0-announce")));
+
+        sortingDropdownList.click();
+
+        WebElement lowToHighDropdownSelection = webDriver.findElement(By.id("s-result-sort-select_1"));
+        lowToHighDropdownSelection.click();
+
+        boolean arePricesInAscendingOrder = verifyPricesAreInAscendingOrder();
+
+        Assert.assertTrue(arePricesInAscendingOrder, "Prices are not in ascending order");
+    }
+
+    private void selectBrand(String brandName) {
+        WebElement brandNameCheckbox = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                .until(ExpectedConditions.presenceOfElementLocated(By.xpath("//span[text() = '" + brandName + "']")));
+        brandNameCheckbox.click();
+    }
+
+
+    private boolean verifyEveryTitleContainsBrandName(String brandName) {
+        WebElement paginationNextBtn;
+        boolean everyTitleContainsInputWord;
+
+        while (true) {
+            paginationNextBtn = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                    .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[contains(@class,'s-pagination-next')]")));
+
+//checking if every title of the page contains brand name
+            everyTitleContainsInputWord = webDriver.findElements(By.xpath("//div[contains(@class,'s-card-container')]//span[contains(@class,'a-size-medium')]"))
```

Comment:
It is better to make this line shorter. Just move some parts to the next line. The same for other